nevergrad/functions/powersystems/core.py

```diff
@@ -0,0 +1,233 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# This code is based on a code and ideas by Emmanuel Centeno and Antoine Moreau,
+# University Clermont Auvergne, CNRS, SIGMA Clermont, Institut Pascal
+
+import copy
+from math import sqrt, tan, pi
+import math
+import matplotlib.pyplot as plt
+from typing import Any
+from typing import List
+import numpy as np
+from nevergrad.common.typetools import ArrayLike
+from ... import instrumentation as inst
+from ...instrumentation.core import Variable
+from ...instrumentation.multivariables import Instrumentation
+pi = np.pi
+
+
+class Agent():
+    """An agent has an input size, an output size, a number of layers, a width of its internal layers 
+    (a.k.a number of neurons per hidden layer)."""
+
+    def __init__(self, input_size: int, output_size: int, layers: int=3, layer_width: int=14):
+        assert(layers >= 2)
+        self.input_size = input_size
+        self.output_size = output_size
+        self.layers: List[Any] = []
+        self.layers += [np.zeros((input_size, layer_width))]
+        for i in range(layers-2):
+            self.layers += [np.zeros((layer_width, layer_width))]
+        self.layers += [np.zeros((layer_width, output_size))]
+        assert len(self.layers) == layers
+
+    def GetParamNumbers(self):
+        return sum([np.prod(l.shape) for l in self.layers])
+
+    def SetParams(self, ww):
+        w = [w for w in ww]
+        assert(len(w) == self.GetParamNumbers())
+        for l in self.layers:
+            s = np.prod(l.shape)
+            l = np.reshape(np.array(w[:s]), l.shape)
+            w = w[s:]
+
+    def GetOutput(self, inp):
+        output = np.array(inp).reshape(1, len(inp))
+        for l in [self.layers[i] for i in range(len(self.layers) - 1)]:
```

Comment:
You can write it more simply as 
`for l in self.layers[:-1]:`

Reply:
Yeah much better :-) done.